src/ORM/DataObject.php

```diff
@@ -1345,7 +1345,7 @@ public function populateDefaults()
      *
      * @return ValidationException Exception generated by this write, or null if valid
      */
-    protected function validateWrite()
+    protected function validateWrite(bool $skipValidation = false)
```

Comment:
This is looks super weird. The method is to validate that a write would occur ... then there's parameter to make it so the method doesn't really do anything.

I get there's the ObsoleteClassName check if you pass in true ... but that's throwing a `ValidationException` and we just said 'skip validation'. It's really confusing

Seems much better to simply not call this method in the first place rather than have a parameter

Reply:
We're skipping validation of data. We can't write anything at all if the `ObsoleteClassName` check fails - that's not really validation in the same sense. That's not validating data, that's really checking if the code is okay.

Reply:
Yeah I guess you're right. We are somewhat bound by the ObsoleteClassName being part of validation though kind of not.